kafka-plugins/src/main/java/co/cask/hydrator/plugin/realtime/KafkaProducer.java

```diff
@@ -0,0 +1,283 @@
+      props.put(ACKS_REQUIRED, "1");
+      isAsync = true;
+    }
+    
+    //config = new ProducerConfig(props);
+    producer = new org.apache.kafka.clients.producer.KafkaProducer<String, String>(props);
+    
+  }
+  
+  @Override
+  public int write(Iterable<StructuredRecord> objects, final DataWriter dataWriter) throws Exception {
+    int count = 0;
+
+    List<Schema.Field> fields = Lists.newArrayList();
+    
+    // For each object
+    for (StructuredRecord object : objects) {
+      
+      // Extract the field names from the object passed in. This is required
+      // because this information is not available in initialize or configuration phase.
+      if(!fieldsExtracted) {
+        fields = object.getSchema().getFields();
+        fieldsExtracted = true;
+      }
+      
+      // Depending on the configuration create a body that needs to be 
+      // built and pushed to Kafka. 
+      String body = "";
+      if (sconfig.format.equalsIgnoreCase("JSON")) {
+        body = StructuredRecordStringConverter.toJsonString(object);
+      } else {
+        // Extract all values from the structured record
+        List<Object> objs = Lists.newArrayList();
+        for(Schema.Field field : fields) {
+          objs.add(object.get(field.getName()));
+        }
+
+        StringWriter writer = new StringWriter();
+        CSVPrinter printer = null;
+        CSVFormat csvFileFormat;
+        switch(sconfig.format.toLowerCase()) {
+          case "csv":
+            csvFileFormat = CSVFormat.Predefined.Default.getFormat();
+            printer = new CSVPrinter(writer, csvFileFormat);
+            break;
+          
+          case "excel":
+            csvFileFormat = CSVFormat.Predefined.Excel.getFormat();
+            printer = new CSVPrinter(writer, csvFileFormat);
+            break;
+          
+          case "mysql":
+            csvFileFormat = CSVFormat.Predefined.MySQL.getFormat();
+            printer = new CSVPrinter(writer, csvFileFormat);
+            break;
+          
+          case "tdf":
+            csvFileFormat = CSVFormat.Predefined.TDF.getFormat();
+            printer = new CSVPrinter(writer, csvFileFormat);
+            break;
+          
+          case "rfc4180":
+            csvFileFormat = CSVFormat.Predefined.TDF.getFormat();
+            printer = new CSVPrinter(writer, csvFileFormat);
+            break;
+        }
+        
+        if (printer != null) {
+          printer.printRecord(objs);
+          body = writer.toString();
+        }
+      }
+      
+      // Message key.
+      String key = "no_key";
+      if(sconfig.key != null) {
+        key = object.get(sconfig.key);    
+      }
+      
+      // Extract the partition key from the record. If the partition key is 
+      // Integer then we use it as-is else
+      Integer partitionKey = 0;
+      if(sconfig.partitionField != null) {
+        if(object.get(sconfig.partitionField).getClass().isInstance(Integer.class)) {
```

Comment:
The only thing you might want to guard against is the object doesn't have the partitionField, which will give you NPE if try to invoke `hashCode`.